adminSiteClient/VariablesAnnotationPage.tsx

```diff
@@ -0,0 +1,927 @@
+            .with(EditorOption.mappingEditor, (_) => "text")
+            .with(EditorOption.primitiveListEditor, (_) => "text")
+            .exhaustive()
+    }
+    static fieldDescriptionToColumn(desc: FieldDescription): JSX.Element {
+        const name = desc.pointer //.substring(1)
+        return (
+            <HotColumn
+                key={desc.pointer}
+                settings={{
+                    title: name,
+                    readOnly:
+                        desc.editor === EditorOption.primitiveListEditor ||
+                        desc.editor === EditorOption.mappingEditor ||
+                        desc.editor === EditorOption.colorEditor,
+                    type: VariablesAnnotationComponent.decideHotType(desc),
+                    source: desc.enumOptions,
+                    data: desc.pointer,
+                    width: Math.max(Bounds.forText(name).width, 50),
+                }}
+            />
+        )
+    }
+    static readOnlyColumnNamesFields = [
+        ["Id", "id"],
+        ["Variable name", "name"],
+        ["Dataset name", "datasetname"],
+        ["Namespace", "namespacename"],
+    ] as const
+
+    @computed get hotColumns(): JSX.Element[] {
+        const { fieldDescriptions } = this
+
+        if (fieldDescriptions === undefined) return []
+        else {
+            // First the 4 readonly columns
+            const readOnlyColumns: JSX.Element[] =
+                VariablesAnnotationComponent.readOnlyColumnNamesFields.map(
+                    (nameAndField) => (
+                        <HotColumn
+                            key={nameAndField[0]}
+                            settings={{
+                                title: nameAndField[0],
+                                readOnly: true,
+                                data: nameAndField[1],
+                                width: Math.max(
+                                    Bounds.forText(nameAndField[0]).width,
+                                    50
+                                ),
+                            }}
+                        />
+                    )
+                )
+            // then all the others coming from the fieldDescriptions
+            // for these we want to suppress the HIDDEN_COLUMNS and
+            // move the IMPORTANT_COLUMNS to the front
+            const fieldDescsFiltered = fieldDescriptions.filter(
+                (desc) =>
+                    HIDDEN_COLUMNS.find((item) => item === desc.pointer) ===
+                    undefined
+            )
+            const fieldDescsMap = new Map(
+                fieldDescsFiltered.map((desc) => [desc.pointer, desc])
+            )
+            const importantDescs = IMPORTANT_COLUMNS.map((pointer) =>
+                fieldDescsMap.get(pointer)
+            )
+            const remainingDescs = differenceOfSets([
+                new Set(fieldDescsMap.keys()),
+                new Set(IMPORTANT_COLUMNS),
```

Comment:
You can actually incorporate the `HIDDEN_COLUMNS` filtering into this call here.